### PR TITLE
Equality Operator - Compare to Symbols and Integers

### PR DESCRIPTION
**Pull request by CyborgMaster**
The `==` operator should behave sanely when comparing to symbols and integers.  

With the following class:
``` ruby
class Patient < ActiveRecord::Base
  extend Enumerize
  enumerize :status, in: { inactive: 0, active: 1 }, scope: true
end
```

I can currently run:
``` ruby
Patient.first.status == 'inactive'
```

But will never work and always return false:
```ruby
Patient.first.status == :inactive
Patient.first.status == 0
```

I can assign a enumerized field a symbol (`Patient.first.status = :inactive`) or an integer (`Patient.first.status = 0`), so I should be able to compare against them as well.

This patch allows comparing to symbols and integers.

### CODE REVIEW

**Comment by CyborgMaster** [comment]
@lest, as long as you are reviewing pull requests, what do you think about this one?

File: lib/enumerize/value.rb [hunk 1]
```diff
@@ -17,6 +17,13 @@ def text
       I18n.t(i18n_keys[0], :default => i18n_keys[1..-1])
     end
 
+    def ==(other)
+      return super(other) if other.is_a? String
+      return super(other.to_s) if other.is_a? Symbol
+      return value == other if other.is_a?(Integer) && value.is_a?(Integer)
+      super other
```
Comment (lest) [inline lib/enumerize/value.rb]: It seems to me that the `==` implementation can be simplified (IMO it has too many `is_a?` checks).

I guess that `super(other.to_s) || value == other` should work.

**Comment by CyborgMaster** [comment]
@lest, you make a very good point.  Changed exactly as you suggested.  And since I have already written tests for this one, I know that your implementation works.

**Comment by lest** [comment]
Thanks!

**Comment by CyborgMaster** [comment]
Awesome!!! Thanks so much @lest.  I've been maintaining these PR's for over a year, its great to finally get them merged.  The only thing that I need now in order to delete my fork entirely is a new release to RubyGems.  Any chance we could get that as well?

**Comment by lest** [comment]
@CyborgMaster Thanks for your contributions! Sorry that I haven't provided feedback in a long time.

I need to test the latest master branch on some rails apps I'm currently working on before making a new release. I can probably do it this week.

**Comment by CyborgMaster** [comment]
No problem at all.  I'm just glad to get them merged now.  

Sounds good with the testing, I would do the same.  Sometime this week is plenty fast for me.  If you could ping me here, or let me know where else I could subscribe to get notified of the new release when it happens, I would appreciate it.

**Comment by lest** [comment]
@CyborgMaster The version 0.10.0 have been released. Thanks!

**Comment by CyborgMaster** [comment]
My gemfile is clean!  No more need to reference my github fork! Yay! 

Thanks again.

**Pull request merged**